desk/app/expose.hoon

```diff
@@ -0,0 +1,2233 @@
+::  exposé: clearweb content rendering
+::
+::    copy a reference to a notebook post, then:
+::    :expose [%show /that/reference/with/id/numbers/quoted/'123456789']
+::    then visit in the browser:
+::    /expose/that/reference/as/copied/123456789
+::
+/-  c=cite, d=channels, co=contacts
+/+  sigil, u=channel-utils,
+    dbug, verb
+::
+|%
++$  state-0
+  $:  %0
+      open=(set cite:c)  ::TODO  could support ranges of msgs?
+  ==
+::
++$  action
+  $%  [%show =path]
+      [%hide =path]
+  ==
+::
++$  card  card:agent:gall
+::
+++  e
+  |%
+  ++  render-post
+    |=  [our=@p now=@da]
+    |=  [=nest:g:c msg=post:d]
+    ^-  (unit manx)
+    =/  aco=(unit contact:co)
+      =/  base=path  /(scot %p our)/contacts/(scot %da now)
+      ?.  .^(? %gu (weld base /$))
+        ~
+      =+  .^(rol=rolodex:co %gx (weld base /all/contact-rolodex))
+      ?~  for=(~(get by rol) author.msg)
+        ~
+      ?.  ?=([[@ ^] *] u.for)
+        ~
+      `con.for.u.for
+    ::
+    ::TODO  if we render replies then we can "unroll" whole chat threads too (:
+    ::TODO  just key off the kind-data, no?
+    |^  ?+  p.nest  ~
+            %chat
+          ?>  ?=(%chat -.kind-data.msg)
+          =/  title=tape
+            (trip (rap 3 (turn (first-inline:u content.msg) flatten-inline:u)))
```

Comment:
I think this is more worthy of being the meta description. Title should be something with group/channel title combo with maybe author as well?